core/src/consensus/tendermint/mod.rs

```diff
@@ -534,6 +544,7 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
             consensus_view(header).expect("This block is generated by this machine."),
             header
         );
+        self.to_step(Step::Prevote);
```

Comment:
When is `proposal_generated` called?
Is it a function that called on the block proposer?
If then, I think you should describe it on the commit message.

Reply:
It should be described on the commit message since it changes the behavior.
The further reader will expect this commit only changes the behavior when the proposal block arrives, but it also changes the behavior when the proposal block is generated.

Reply:
Oh, I see. I wrote "when proposal block **arrives**". I'll fix this.
Also, I think there should be some comments for readers. It's too hard to figure out where those callbacks are called. Even I added less than 100 lines of code, it takes too much time :(